xWebVirtualDirectory: Must specify '/' for WebApplication when virtual directory is under the website

Hello,

When running the resource with 'Absent' flag, I'm getting the following error:
`##[error]PowerShell Desired State Configuration does not support execution of commands in an interactive mode. Please ensure 
that the underlying command is not prompting for user input, such as missing mandatory parameter, confirmation prompt 
etc.
    + CategoryInfo          : NotSpecified: (:) [], CimException
    + FullyQualifiedErrorId : System.NotSupportedException,Microsoft.IIs.PowerShell.Provider.RemoveVirtualDirectoryCom 
   mand
    + PSComputerName        : IPAddress`

'Ensure' works perfectly but when I run it with 'Absent' it brings up the above error.

Appreciate your assistance,
David

Hello johlju,

Thank you for your quick response.

Note that the problematic is **Vd1**.

**Verbose output of the xWebVirtualDirectory section**
```
2018-06-10T14:47:45.0167701Z VERBOSE: [ServerName]: LCM:  [ Start  Resource ]  [[xWebVirtualDirectory]Vd1]
2018-06-10T14:47:45.0167701Z VERBOSE: [ServerName]: LCM:  [ Start  Test     ]  [[xWebVirtualDirectory]Vd1]
2018-06-10T14:47:45.0636518Z VERBOSE: [ServerName]: LCM:  [ End    Test     ]  [[xWebVirtualDirectory]Vd1]  in 0.0780 seconds.
2018-06-10T14:47:45.0636518Z VERBOSE: [ServerName]: LCM:  [ Start  Set      ]  [[xWebVirtualDirectory]Vd1]
2018-06-10T14:47:45.1104968Z VERBOSE: [ServerName]:                            [[xWebVirtualDirectory]Vd1] Removing existing Virtual Directory "Vd1".
2018-06-10T14:47:45.4386355Z ##[error]PowerShell Desired State Configuration does not support execution of commands in an interactive mode. Please ensure 
that the underlying command is not prompting for user input, such as missing mandatory parameter, confirmation prompt 
etc.
    + CategoryInfo          : NotSpecified: (:) [], CimException
    + FullyQualifiedErrorId : System.NotSupportedException,Microsoft.IIs.PowerShell.Provider.RemoveVirtualDirectoryCom 
   mand
    + PSComputerName        : x.x.x.x
 

2018-06-10T14:47:45.4542634Z VERBOSE: [ServerName]: LCM:  [ End    Set      ]  [[xWebVirtualDirectory]Vd1]  in 0.1870 seconds.
2018-06-10T14:47:45.4698911Z ##[error]The PowerShell DSC resource '[xWebVirtualDirectory]i' with SourceInfo 
'D:\PathToScript\WebSite.ps1::804::13::xWebVirtualDirectory' threw one or more 
non-terminating errors while running the Set-TargetResource functionality. These errors are logged to the ETW channel 
called Microsoft-Windows-DSC/Operational. Refer to this channel for more details.
    + CategoryInfo          : InvalidOperation: (:) [], CimException
    + FullyQualifiedErrorId : NonTerminatingErrorFromProvider
    + PSComputerName        : x.x.x.x
 

2018-06-10T14:47:45.4698911Z VERBOSE: [ServerName]: LCM:  [ Start  Resource ]  [[xWebVirtualDirectory]Vd2]
2018-06-10T14:47:45.4698911Z VERBOSE: [ServerName]: LCM:  [ Start  Test     ]  [[xWebVirtualDirectory]Vd2]
2018-06-10T14:47:45.4698911Z VERBOSE: [ServerName]:                            [[xWebVirtualDirectory]Vd2] Physical path 
2018-06-10T14:47:45.4698911Z "\\ServerName\PathToVd2" for web virtual directory "Vd2" does not match desired
2018-06-10T14:47:45.4698911Z  state.
2018-06-10T14:47:45.4698911Z VERBOSE: [ServerName]: LCM:  [ End    Test     ]  [[xWebVirtualDirectory]Vd2]  in 0.0470 seconds.
2018-06-10T14:47:45.4698911Z VERBOSE: [ServerName]: LCM:  [ Start  Set      ]  [[xWebVirtualDirectory]Vd2]
2018-06-10T14:47:45.4698911Z VERBOSE: [ServerName]:                            [[xWebVirtualDirectory]Vd2] Updating physical path for web virtual 
2018-06-10T14:47:45.4698911Z directory "Vd2".
2018-06-10T14:47:45.4698911Z VERBOSE: [ServerName]: LCM:  [ End    Set      ]  [[xWebVirtualDirectory]Vd2]  in 0.1090 seconds.
2018-06-10T14:47:45.4698911Z VERBOSE: [ServerName]: LCM:  [ End    Resource ]  [[xWebVirtualDirectory]Vd2]
2018-06-10T14:47:45.4855009Z VERBOSE: [ServerName]: LCM:  [ Start  Resource ]  [[xWebVirtualDirectory]Vd3]
2018-06-10T14:47:45.4855009Z VERBOSE: [ServerName]: LCM:  [ Start  Test     ]  [[xWebVirtualDirectory]Vd3]
2018-06-10T14:47:45.5480256Z VERBOSE: [ServerName]:                            [[xWebVirtualDirectory]Vd3] Web virtual directory is in required 
2018-06-10T14:47:45.5480256Z state.
2018-06-10T14:47:45.5480256Z VERBOSE: [ServerName]: LCM:  [ End    Test     ]  [[xWebVirtualDirectory]Vd3]  in 0.0630 seconds.
2018-06-10T14:47:45.5480256Z VERBOSE: [ServerName]: LCM:  [ Skip   Set      ]  [[xWebVirtualDirectory]Vd3]
2018-06-10T14:47:45.5480256Z VERBOSE: [ServerName]: LCM:  [ End    Resource ]  [[xWebVirtualDirectory]Vd3]
2018-06-10T14:47:45.5636398Z VERBOSE: [ServerName]: LCM:  [ Start  Resource ]  [[xWebVirtualDirectory]vd4]
2018-06-10T14:47:45.5636398Z VERBOSE: [ServerName]: LCM:  [ Start  Test     ]  [[xWebVirtualDirectory]vd4]
2018-06-10T14:47:45.5949062Z VERBOSE: [ServerName]:                            [[xWebVirtualDirectory]vd4] Web virtual directory is in required 
2018-06-10T14:47:45.5949062Z state.
2018-06-10T14:47:45.5949062Z VERBOSE: [ServerName]: LCM:  [ End    Test     ]  [[xWebVirtualDirectory]vd4]  in 0.0620 seconds.
2018-06-10T14:47:45.5949062Z VERBOSE: [ServerName]: LCM:  [ Skip   Set      ]  [[xWebVirtualDirectory]vd4]
2018-06-10T14:47:45.5949062Z VERBOSE: [ServerName]: LCM:  [ End    Resource ]  [[xWebVirtualDirectory]vd4]
2018-06-10T14:47:45.6105215Z VERBOSE: [ServerName]: LCM:  [ Start  Resource ]  [[xWebVirtualDirectory]vd5]
2018-06-10T14:47:45.6105215Z VERBOSE: [ServerName]: LCM:  [ Start  Test     ]  [[xWebVirtualDirectory]vd5]
2018-06-10T14:47:45.6417822Z VERBOSE: [ServerName]:                            [[xWebVirtualDirectory]vd5] Web virtual directory is in required 
2018-06-10T14:47:45.6417822Z state.
2018-06-10T14:47:45.6417822Z VERBOSE: [ServerName]: LCM:  [ End    Test     ]  [[xWebVirtualDirectory]vd5]  in 0.0470 seconds.
2018-06-10T14:47:45.6417822Z VERBOSE: [ServerName]: LCM:  [ Skip   Set      ]  [[xWebVirtualDirectory]vd5]
2018-06-10T14:47:45.6417822Z VERBOSE: [ServerName]: LCM:  [ End    Resource ]  [[xWebVirtualDirectory]vd5]
2018-06-10T14:47:45.6730704Z VERBOSE: [ServerName]: LCM:  [ Start  Resource ]  [[xWebVirtualDirectory]vd6]
2018-06-10T14:47:45.6730704Z VERBOSE: [ServerName]: LCM:  [ Start  Test     ]  [[xWebVirtualDirectory]vd6]
2018-06-10T14:47:45.6886510Z VERBOSE: [ServerName]:                            [[xWebVirtualDirectory]vd6] Web virtual directory is in required 
2018-06-10T14:47:45.6886510Z state.
2018-06-10T14:47:45.7042641Z VERBOSE: [ServerName]: LCM:  [ End    Test     ]  [[xWebVirtualDirectory]vd6]  in 0.0470 seconds.
2018-06-10T14:47:45.7042641Z VERBOSE: [ServerName]: LCM:  [ Skip   Set      ]  [[xWebVirtualDirectory]vd6]
2018-06-10T14:47:45.7042641Z VERBOSE: [ServerName]: LCM:  [ End    Resource ]  [[xWebVirtualDirectory]vd6]
2018-06-10T14:47:45.7198852Z VERBOSE: [ServerName]: LCM:  [ Start  Resource ]  [[xWebVirtualDirectory]vd7]
2018-06-10T14:47:45.7198852Z VERBOSE: [ServerName]: LCM:  [ Start  Test     ]  [[xWebVirtualDirectory]vd7]
2018-06-10T14:47:45.7511513Z VERBOSE: [ServerName]:                            [[xWebVirtualDirectory]vd7] Web virtual directory is in required state.
2018-06-10T14:47:45.7511513Z VERBOSE: [ServerName]: LCM:  [ End    Test     ]  [[xWebVirtualDirectory]vd7]  in 0.0470 seconds.
2018-06-10T14:47:45.7511513Z VERBOSE: [ServerName]: LCM:  [ Skip   Set      ]  [[xWebVirtualDirectory]vd7]
2018-06-10T14:47:45.7511513Z VERBOSE: [ServerName]: LCM:  [ End    Resource ]  [[xWebVirtualDirectory]vd7]
2018-06-10T14:47:45.7667672Z VERBOSE: [ServerName]: LCM:  [ Start  Resource ]  [[xWebVirtualDirectory]vd8]
2018-06-10T14:47:45.7667672Z VERBOSE: [ServerName]: LCM:  [ Start  Test     ]  [[xWebVirtualDirectory]vd8]
2018-06-10T14:47:45.7980302Z VERBOSE: [ServerName]:                            [[xWebVirtualDirectory]vd8] Web virtual directory is in required 
2018-06-10T14:47:45.7980302Z state.
2018-06-10T14:47:45.7980302Z VERBOSE: [ServerName]: LCM:  [ End    Test     ]  [[xWebVirtualDirectory]vd8]  in 0.0470 seconds.
2018-06-10T14:47:45.7980302Z VERBOSE: [ServerName]: LCM:  [ Skip   Set      ]  [[xWebVirtualDirectory]vd8]
2018-06-10T14:47:45.7980302Z VERBOSE: [ServerName]: LCM:  [ End    Resource ]  [[xWebVirtualDirectory]vd8]
2018-06-10T14:47:45.8136493Z VERBOSE: [ServerName]: LCM:  [ Start  Resource ]  [[Script]SetVirtualDirectoriesDefaults]
2018-06-10T14:47:45.8136493Z VERBOSE: [ServerName]: LCM:  [ Start  Test     ]  [[Script]SetVirtualDirectoriesDefaults]
2018-06-10T14:47:45.8605246Z VERBOSE: [ServerName]: LCM:  [ End    Test     ]  [[Script]SetVirtualDirectoriesDefaults]  in 0.0630 seconds.
2018-06-10T14:47:45.8605246Z VERBOSE: [ServerName]: LCM:  [ Skip   Set      ]  [[Script]SetVirtualDirectoriesDefaults]
2018-06-10T14:47:45.8605246Z VERBOSE: [ServerName]: LCM:  [ End    Resource ]  [[Script]SetVirtualDirectoriesDefaults]
2018-06-10T14:47:45.8605246Z VERBOSE: [ServerName]: LCM:  [ End    Set      ]
2018-06-10T14:47:45.9386357Z ##[error]The SendConfigurationApply function did not succeed.
    + CategoryInfo          : NotSpecified: (root/Microsoft/...gurationManager:String) [], CimException
    + FullyQualifiedErrorId : MI RESULT 1
    + PSComputerName        : x.x.x.x
 

2018-06-10T14:47:45.9386357Z VERBOSE: Operation 'Invoke CimMethod' complete.
2018-06-10T14:47:45.9386357Z VERBOSE: Time taken for configuration job to complete is 116.738 seconds
2018-06-10T14:47:45.9386357Z 
```

This is the configuration section:

```
xWebVirtualDirectory Vd1
{
	Name = 'Vd1'
	Website = $Node.WebSiteName
	WebApplication = ''
	Ensure = 'Absent'
	DependsOn = '[xWebsite]Web'
}

xWebVirtualDirectory Vd2
{
	Name = 'Vd2'
	Website = $Node.WebSiteName
	WebApplication = ''
	PhysicalPath = "\\ServerName\PathToVD\Vd2"
	Ensure = 'Present'
	DependsOn = '[xWebsite]Web'
}

xWebVirtualDirectory Vd3
{
	Name = 'Vd3'
	Website = $Node.WebSiteName
	WebApplication = ''
	PhysicalPath = "\\ServerName\PathToVD\Vd3"
	Ensure = 'Present'
	DependsOn = '[xWebsite]Web'
}

xWebVirtualDirectory Vd4
{
	Name = 'Vd4'
	Website = $Node.WebSiteName
	WebApplication = ''
	PhysicalPath = "\\ServerName\PathToVD\Vd4"
	Ensure = 'Present'
	DependsOn = '[xWebsite]Web'
}

xWebVirtualDirectory Vd5
{
	Name = 'Vd5'
	Website = $Node.WebSiteName
	WebApplication = ''
	PhysicalPath = "\\ServerName\PathToVD\Vd5"
	Ensure = 'Present'
	DependsOn = '[xWebsite]Web'
}

xWebVirtualDirectory Vd6
{
	Name = 'Vd6'
	Website = $Node.WebSiteName
	WebApplication = ''
	PhysicalPath = "\\ServerName\PathToVD\Vd6"
	Ensure = 'Present'
	DependsOn = '[xWebsite]Web'
}

xWebVirtualDirectory Vd7
{
	Name = 'Vd7'
	Website = $Node.WebSiteName
	WebApplication = ''
	PhysicalPath = "\\ServerName\PathToVD\Vd7"
	Ensure = 'Present'
	DependsOn = '[xWebsite]Web'
}

xWebVirtualDirectory Vd8
{
	Name = 'Vd8'
	Website = $Node.WebSiteName
	WebApplication = ''
	PhysicalPath = "\\ServerName\PathToVD\Vd8"
	Ensure = 'Present'
	DependsOn = '[xWebsite]Web'
}

Script SetVirtualDirectoriesDefaults
{
	DependsOn = "[xWebsite]Web"
	TestScript = {
		$WebSite = Get-Website -Name $using:Node.WebSiteName
		If (($WebSite.virtualDirectoryDefaults.userName -ne "") -and ($WebSite.virtualDirectoryDefaults.password -ne "")) {
			return $true
		}
		else {
			return $false
		}
	}
	GetScript = {
		# Do Nothing
	}
	SetScript = {
		$WebSiteName = $using:Node.WebSiteName
		$WebSitePath = "IIS:\Sites\"+$WebSiteName
		$WebSite = Get-Item $WebSitePath
		$WebSite.virtualDirectoryDefaults.userName = ******************
		$WebSite.virtualDirectoryDefaults.password = ******************
		$WebSite | Set-Item
	}
} 

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,14 @@ For older change log history see the [historic changelog](HISTORIC_CHANGELOG.md)
 
 ## [Unreleased]
 
+### Fixed
+
+- WebVirtualDirectory
+  [Issue #366](https://github.com/dsccommunity/WebAdministrationDsc/issues/366)
+  In WebVirtualDirectory WebApplication '' and '/' can now be used interchangeably.
+  - Fixed Add WebVirtualDirectory when WebApplication = '/'.
+  - Fixed Remove WebVirtualDirectory when WebApplication = ''.
+
 ## [4.0.0] - 2022-09-17
 
 ### Changed

--- a/source/DSCResources/DSC_WebVirtualDirectory/DSC_WebVirtualDirectory.psm1
+++ b/source/DSCResources/DSC_WebVirtualDirectory/DSC_WebVirtualDirectory.psm1
@@ -102,6 +102,19 @@ function Set-TargetResource
 
     if ($Ensure -eq 'Present')
     {
+        <#
+            Issue #366
+            WebApplication = '/' will cause New-WebVirtualDirectory to write
+            double slash ('//$Name') to config file.
+            This in turn causes Get-WebVirtualDirectory to not find the Virtual Directory.
+            WebApplication = '' works.
+            Note the opposite problem with Remove-WebVirtualDirectory.
+        #>
+        if ($WebApplication -eq '/')
+        {
+            $WebApplication = ''
+        }
+
         $virtualDirectory = Get-WebVirtualDirectory -Site $Website `
                                                     -Name $Name `
                                                     -Application $WebApplication
@@ -134,6 +147,18 @@ function Set-TargetResource
 
     if ($Ensure -eq 'Absent')
     {
+        <#
+            Issue #366
+            WebApplication = '' will cause Remove-WebVirtualDirectory to throw
+            "PowerShell Desired State Configuration does not support execution of commands in an interactive mode ...".
+            WebApplication = '/' works.
+            Note the opposite problem with New-WebVirtualDirectory.
+        #>
+        if ($WebApplication -eq '')
+        {
+            $WebApplication = '/'
+        }
+
         Write-Verbose -Message ($script:localizedData.VerboseSetTargetRemoveVirtualDirectory -f $Name)
         Remove-WebVirtualDirectory -Site $Website `
                                    -Application $WebApplication `

--- a/tests/Integration/DSC_WebVirtualDirectory.Integration.Tests.ps1
+++ b/tests/Integration/DSC_WebVirtualDirectory.Integration.Tests.ps1
@@ -88,6 +88,48 @@ try
             $result.path            | Should Be "/$($DSCConfig.AllNodes.WebVirtualDirectory)"
             $result.physicalPath    | Should Be $DSCConfig.AllNodes.PhysicalPath
         }
+
+        It 'Should create a WebVirtualDirectory with WebApplication = ''/''' -Test {
+
+            configuration DSC_WebVirtualDirectory_WebApplicationSlash
+            {
+                Import-DscResource -ModuleName WebAdministrationDsc
+
+                Node $AllNodes.NodeName
+                {
+                    WebVirtualDirectory WebVirtualDirectory
+                    {
+                        Ensure = 'Present'
+                        Website = $Node.Website
+                        WebApplication = '/'
+                        Name = $Node.WebVirtualDirectory
+                        PhysicalPath = $Node.PhysicalPath
+                    }
+                }
+            }
+
+            & "DSC_WebVirtualDirectory_WebApplicationSlash" `
+            -OutputPath $TestDrive `
+            -ConfigurationData $dscConfig
+
+            Reset-DscLcm
+
+            Start-DscConfiguration `
+                -Path $TestDrive `
+                -ComputerName localhost `
+                -Wait `
+                -Verbose `
+                -Force `
+                -ErrorAction Stop
+
+            # Build results to test
+            $result = Get-WebVirtualDirectory -Site $DSCConfig.AllNodes.Website `
+                -Application '/' `
+                -Name $DSCConfig.AllNodes.WebVirtualDirectory
+
+            # Test virtual directory settings are correct
+            $result                 | Should Not BeNullOrEmpty
+        }
     }
 
     Describe "$($script:dscResourceName)_Absent" {
@@ -138,6 +180,99 @@ try
             $result = Get-WebVirtualDirectory -Site $DSCConfig.AllNodes.Website `
                 -Application $DSCConfig.AllNodes.WebApplication `
                 -Name $DSCConfig.AllNodes.WebVirtualDirectory
+
+            # Test virtual directory is removed
+            $result | Should BeNullOrEmpty
+        }
+
+        It 'Should remove a WebVirtualDirectory with WebApplication = ''''' -Test {
+            # Avoid collision with other tests
+            $virtualDirectoryName = "$($Node.WebVirtualDirectory)2"
+
+            # Declare local configurations
+            configuration DSC_WebVirtualDirectory_WebApplicationBlank_add
+            {
+                Import-DscResource -ModuleName WebAdministrationDsc
+
+                Node $AllNodes.NodeName
+                {
+                    WebVirtualDirectory WebVirtualDirectory
+                    {
+                        Ensure = 'Present'
+                        Website = $Node.Website
+                        WebApplication = ''
+                        Name = $virtualDirectoryName
+                        PhysicalPath = $Node.PhysicalPath
+                    }
+                }
+            }
+
+            configuration DSC_WebVirtualDirectory_WebApplicationBlank_remove
+            {
+                Import-DscResource -ModuleName WebAdministrationDsc
+
+                Node $AllNodes.NodeName
+                {
+                    WebVirtualDirectory WebVirtualDirectory
+                    {
+                        Ensure = 'Absent'
+                        Website = $Node.Website
+                        WebApplication = ''
+                        Name = $virtualDirectoryName
+                        PhysicalPath = $Node.PhysicalPath
+                    }
+                }
+            }
+
+            # local helper
+            function Get-WebApplicationBlankVirtualDirectory()
+            {
+                return Get-WebVirtualDirectory -Site $DSCConfig.AllNodes.Website `
+                    -Application '' `
+                    -Name $virtualDirectoryName
+            }
+
+            # Execute setup
+            & "DSC_WebVirtualDirectory_WebApplicationBlank_add" `
+            -OutputPath $TestDrive `
+            -ConfigurationData $dscConfig
+
+            Reset-DscLcm
+
+            Start-DscConfiguration `
+                -Path $TestDrive `
+                -ComputerName localhost `
+                -Wait `
+                -Verbose `
+                -Force `
+                -ErrorAction Stop
+
+            # Verify intermediate result
+            $resultIntermediate = Get-WebApplicationBlankVirtualDirectory
+
+            # Virtual directory have been created
+            $resultIntermediate     | Should Not BeNullOrEmpty
+
+            # Execute Test operation
+            & "DSC_WebVirtualDirectory_WebApplicationBlank_remove" `
+            -OutputPath $TestDrive `
+            -ConfigurationData $dscConfig
+
+            <#
+                Issue #366
+                Before change this statement throws exception
+                "PowerShell Desired State Configuration does not support execution of commands in an interactive mode ..."
+            #>
+            Start-DscConfiguration `
+                -Path $TestDrive `
+                -ComputerName localhost `
+                -Wait `
+                -Verbose `
+                -Force `
+                -ErrorAction Stop
+
+            # Build results to test
+            $result = Get-WebApplicationBlankVirtualDirectory
 
             # Test virtual directory is removed
             $result | Should BeNullOrEmpty

--- a/tests/Unit/DSC_WebVirtualDirectory.Tests.ps1
+++ b/tests/Unit/DSC_WebVirtualDirectory.Tests.ps1
@@ -186,6 +186,30 @@ try
                 }
             }
 
+            Context 'Ensure = Present and WebApplication = ''/''' {
+                # Issue #366
+                It 'Should change WebApplication to ''''' {
+                    $mockSite = @{
+                        Name = 'SomeName'
+                        Website = 'Website'
+                        WebApplication = '/'
+                        PhysicalPath = 'PhysicalPath'
+                    }
+
+                    Mock -CommandName New-WebVirtualDirectory -MockWith { return $null }
+
+                    Set-TargetResource -Website $mockSite.Website `
+                        -WebApplication $mockSite.WebApplication `
+                        -Name $mockSite.Name `
+                        -PhysicalPath $mockSite.PhysicalPath `
+                        -Ensure 'Present'
+
+                    Assert-MockCalled -CommandName New-WebVirtualDirectory -Exactly 1 -ParameterFilter {
+                        return "$Application" -eq ''
+                    }
+                }
+            }
+
             Context 'Ensure = Present and virtual directory exists' {
                 It 'Should call Set-ItemProperty' {
                     $mockSite = @{
@@ -228,6 +252,31 @@ try
                         -Ensure 'Absent'
 
                     Assert-MockCalled -CommandName Remove-WebVirtualDirectory -Exactly 1
+                }
+            }
+
+            Context 'Ensure = Absent and WebApplication = ''''' {
+                # Issue #366
+                It 'Should change WebApplication to ''/''' {
+                    $mockSite = @{
+                        Name = 'SomeName'
+                        Website = 'Website'
+                        WebApplication = ''
+                        PhysicalPath = 'PhysicalPath'
+                        Count = 1
+                    }
+
+                    Mock -CommandName Remove-WebVirtualDirectory -MockWith { return $null }
+
+                    Set-TargetResource -Website $mockSite.Website `
+                        -WebApplication $mockSite.WebApplication `
+                        -Name $mockSite.Name `
+                        -PhysicalPath $mockSite.PhysicalPath `
+                        -Ensure 'Absent'
+
+                    Assert-MockCalled -CommandName Remove-WebVirtualDirectory -Exactly 1 -ParameterFilter {
+                        return "$Application" -eq '/'
+                    }
                 }
             }
         }

